common/include/villas/memory.hpp

```diff
@@ -249,10 +249,12 @@ class HostRam {
     allocateBlock(size_t size);
   };
 
-  static HostRamAllocator &getAllocator() { return allocator; }
-
-private:
-  static HostRamAllocator allocator;
+  static HostRamAllocator &getAllocator() {
```

Comment:
Could we use the same approach for the logging class as well?

E.g. have a static function, with a static private/projected member as singleton?

```cpp
Log::getLogger("some_name")
```

is nicer than:

```cpp
logging().get("some_name")
```

I also prefer to have an harmonized way to create those singletons.

Reply:
ok. I wanted to be minimally invasive, but this sounds sensible.